front/tests/pages/scenario-page-model.ts

```diff
@@ -1,321 +1,170 @@
-      .getByTestId('allowances-standard-settings-value-input');
-    this.getAllowancesEnergyConsumed = page.getByTestId('allowance-energy-consumed');
-    this.getAverageEnergyConsumed = page.getByTestId('average-energy-consumed');
-    this.getAllowancesEngineeringSettings = page
-      .getByTestId('engineering-allowance-group')
-      .last()
-      .getByTestId('allowances-engineering-allowance-input');
-    this.getSuccessBtn = page.getByTestId('add-allowance-button');
-    this.getScenarioUpdateConfirmBtn = page.locator('#modal-content').getByTestId('updateScenario');
-  }
-
-  async openTabByDataId(id: string) {
-    await this.page.getByTestId(id).click();
-  }
-
+      .getByTestId('delete-scenario');
+    this.trainLabels = page.getByTestId('add-train-labels');
+    this.addScenarioButton = page.getByTestId('add-scenario-button');
+    this.scenarioNameInput = page.locator('#scenarioInputName');
+    this.scenarioDescriptionInput = page.locator('#scenarioDescription');
+    this.scenarioInfraList = page.getByTestId('infra-list');
+    this.scenarioElectricProfileSelect = page.locator('.input-group');
+    this.scenarioName = page.locator('.scenario-details-name .scenario-name');
+    this.scenarioDescription = page.locator('.scenario-details-description');
+    this.scenarioInfraName = page.locator('.scenario-infra-name');
+    this.trainEditButton = page.locator('.scenario-timetable-train-buttons-update');
+    this.scenarioConfirmUpdateButton = page
+      .locator('#modal-content')
+      .getByTestId('update-scenario');
+    this.createScenarioButton = page.getByTestId('create-scenario');
+    this.scenarioTagsLabel = page.getByTestId('scenario-details-tag');
+  }
+
+  // Creates a scenario based on the provided details.
+  async createScenario(details: ScenarioDetails) {
+    expect(this.addScenarioButton).toBeVisible();
+    await this.addScenarioButton.click();
+    await this.fillScenarioDetails(details);
+    await this.createScenarioButton.click();
+    await this.page.waitForURL('**/scenarios/*');
+  }
+
+  // Updates a scenario based on the provided details.
+  async updateScenario(details: ScenarioDetails) {
+    await this.clickOnUpdateScenario();
+    await this.fillScenarioDetails(details);
+    await this.scenarioConfirmUpdateButton.click();
+    await this.page.waitForURL('**/scenarios/*');
+  }
+
+  // Fills the scenario details in the form inputs.
+  private async fillScenarioDetails({
+    name,
+    description,
+    infraName,
+    tags,
+    electricProfileName,
+  }: ScenarioDetails) {
+    await this.scenarioNameInput.fill(name);
+    await this.scenarioDescriptionInput.fill(description);
+
+    // Set electric profile if provided
+    if (electricProfileName) {
+      await this.setScenarioElectricProfileByName(electricProfileName);
+    }
+
+    // Select infra name if provided
+    if (infraName) {
+      await this.scenarioInfraList.getByText(infraName).first().click();
+    }
+
```

Comment:
what if not?

Reply:
Both `infraName` and `electricProfileName` are optional in this function, so if they are not provied the action will not be done.